CMake/CMakeExtras.cmake

```
@@ -0,0 +1,21 @@
+# Set of helper functions for our CMake
+
+# Joins arguments and places the results in ${result_var}.
+function(join result_var)
+  set(result )
```

```suggestion
  set(result)
```
This comment is basically the pinnacle of pedantry. :)